decidim-budgets/app/controllers/decidim/budgets/admin/application_controller.rb

```diff
@@ -16,15 +16,15 @@ def budget
         end
 
         def projects
-          return unless budget
-
-          @projects ||= budget.projects
+          @projects ||= if budget
+                          budget.projects
+                        else
+                          Decidim::Budgets::Project.joins(:budget).where(budget: { component: current_component })
+                        end
```

Comment:
This changes the behavior so I would not implement this change.

Reply:
![image](https://github.com/decidim/decidim/assets/105683/6bbc3397-a278-4e20-9873-64342d2db99f)
The scope of this change is to ensure that we always have a valid collection of projects either directly from `budget` object either from the budgets associated to the `current_component`.

Reply:
Or maybe we could move the project attachments and project attackments collection methods inside the budget, when we would also have the budget scope when managing the attachments and attachment collections?